### PR TITLE
remaining junit5

### PR DESCRIPTION
**Pull request by barisoyoruk**
In [this issue](https://github.com/palantir/atlasdb/issues/6796), we are transitioning our test classes from JUnit4 to JUnit5.

In this pull request, we are evaluating `TODO` comments on a case-by-case basis and upgrading `TestRules` to `TestExtension` where appropriate. For more details, see the comments below.

After this PR, there will be a decrease in the number of tests run because test classes that previously appeared in the suite class were being executed twice, even though there was no reason to run them twice.

Decrease due to AtlasConsoleTestRunner -> 29
- Decrease for test-suite-0: 29

### CODE REVIEW

File: atlasdb-autobatch/src/test/java/com/palantir/atlasdb/autobatch/TracingEventHandlerTest.java [hunk 1]
```diff
@@ -18,20 +18,14 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
 
-/* TODO(boyoruk): Do not forget to upgrade this to JUnit5. */
-@RunWith(MockitoJUnitRunner.class)
+@ExtendWith(MockitoExtension.class)
 public class TracingEventHandlerTest {
 
-    @Rule
-    public RenderTracingRule renderTracingRule = new RenderTracingRule();
-
```
Comment (barisoyoruk) [inline atlasdb-autobatch/src/test/java/com/palantir/atlasdb/autobatch/TracingEventHandlerTest.java:34]: This has no effect on the test case. It just writes the tracing output to a file. This line is added before to examine what was going on while we are running the test. However, I think we don't need it anymore.

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/util/DropwizardClientRule.java [hunk 2]
```diff
@@ -1,94 +0,0 @@
...
-import org.junit.rules.ExternalResource;
-
-/* TODO(boyoruk): Migrate to JUnit5 */
-/**
- * Copied from {@link io.dropwizard.testing.junit.DropwizardClientRule} so that we can configure the ObjectMapper.
- */
-public class DropwizardClientRule extends ExternalResource {
-    private final Object[] resources;
-    private final DropwizardTestSupport<Configuration> testSupport;
-
```
Comment (barisoyoruk) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/util/DropwizardClientRule.java:37]: This class seems leftover.

File: timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionTestSetup.java [hunk 3]
```diff
@@ -56,40 +51,27 @@ public final class TimeLockCorruptionTestSetup implements TestRule {
...
-        localTimestampInvariantsVerifier = new LocalTimestampInvariantsVerifier(localDataSource);
+    void setup() {
+        try {
+            File randomFile1 = Files.createTempDirectory("randomFile1").toFile();
+            randomFile1.deleteOnExit();
+            File randomFile2 = Files.createTempDirectory("randomFile2").toFile();
+            randomFile1.deleteOnExit();
+            File randomFile3 = Files.createTempDirectory("randomFile3").toFile();
+            randomFile1.deleteOnExit();
+
```
Comment (barisoyoruk) [inline timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionTestSetup.java:62]: We cannot use @TempDir extension inside another extension. It requires @Test annotation to be injected.

File: atlasdb-console/src/test/groovy/com/palantir/atlasdb/console/groovy/AtlasConsoleTestRunner.groovy [hunk 4]
```diff
@@ -1,26 +0,0 @@
...
- */
-package com.palantir.atlasdb.console.groovy
-
-import org.junit.runner.RunWith
-import org.junit.runners.Suite
-import org.junit.runners.Suite.SuiteClasses
-
-@RunWith(Suite)
-@SuiteClasses([TableTest, AtlasConsoleServiceWrapperTest])
-class AtlasConsoleTestRunner {
```
Comment (barisoyoruk) [inline atlasdb-console/src/test/groovy/com/palantir/atlasdb/console/groovy/AtlasConsoleTestRunner.groovy:24]: Not needed. We were just running the tests twice.

File: timelock-server/build.gradle [hunk 5]
```diff
@@ -76,12 +76,6 @@ dependencies {
     }
     runtimeOnly project(':atlasdb-dbkvs')
 
-    testImplementation 'com.github.tomakehurst:wiremock-standalone'
-    testImplementation 'com.google.guava:guava'
-    testImplementation 'com.palantir.conjure.java.runtime:client-config'
-    testImplementation 'com.palantir.docker.compose:docker-compose-rule-core'
-    testImplementation 'com.palantir.docker.compose:docker-compose-rule-junit4'
-    testImplementation 'io.dropwizard.metrics:metrics-core'
```
Comment (barisoyoruk) [inline timelock-server/build.gradle:84]: All of these were unnecessary for testImplementation.

File: leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java [hunk 6]
```diff
@@ -65,10 +62,7 @@ public class AwaitingLeadershipProxyTest {
     private final Runnable mockRunnable = mock(Runnable.class);
     private final Supplier<Runnable> delegateSupplier = Suppliers.ofInstance(mockRunnable);
 
-    @Rule
-    public final RenderTracingRule rule = new RenderTracingRule();
-
```
Comment (barisoyoruk) [inline leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java:70]: Same reasoning as above

File: timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionDetectionHelper.java [hunk 7]
```diff
@@ -28,17 +28,16 @@
...
-/* TODO(boyoruk): Find a way to implement this in JUnit5. */
-public final class TimeLockCorruptionDetectionHelper implements TestRule {
-    private TimeLockCorruptionTestSetup timeLockCorruptionTestSetup = new TimeLockCorruptionTestSetup();
+public final class TimeLockCorruptionDetectionHelper implements BeforeEachCallback {
+
+    private final TimeLockCorruptionTestSetup timeLockCorruptionTestSetup = new TimeLockCorruptionTestSetup();
 
     @Override
-    public Statement apply(Statement base, Description description) {
-        return timeLockCorruptionTestSetup.apply(base, description);
```
Comment (barisoyoruk) [inline timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionDetectionHelper.java:41]: Imho, this implementation was little bit overkill. It is enough to make either one of `TimeLockCorruptionTestSetup` or `TimeLockCorruptionDetectionHelper` a `TestRule`

File: timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionTestSetup.java (hunk 3 above)
Comment (barisoyoruk) [reply timelock-corruption-detection/src/test/java/com/palantir/timelock/corruption/detection/TimeLockCorruptionTestSetup.java:62]: Also,  

> Extensions should not depend on other extension - not even on those built-in extensions that Jupiter provides. - JUnit developer

https://github.com/junit-team/junit5/issues/1786#issuecomment-464618781

File: versions.props [hunk 8]
```diff
@@ -6,7 +6,7 @@ com.github.ben-manes.caffeine:caffeine = 3.1.8
 com.github.peterwippermann.junit4:parameterized-suite = 1.1.0
 com.github.rholder:guava-retrying = 2.0.0
 com.github.stefanbirkner:system-rules = 1.19.0
-com.github.tomakehurst:wiremock-standalone = 2.27.2
+com.github.tomakehurst:wiremock-standalone = 3.0.1
```
Comment (barisoyoruk) [inline versions.props:9]: The previous version did not support JUnit5.

**Review by LucasIME — APPROVED** [review]

**Pull request merged**